### PR TITLE
Fix blood-red magboots and speed boots sprite toggle issues.

### LINKED ISSUE #28354: speedboots need a clothing sprite for the on state
## Description
the icon has an on state but not the actual in world clothing sprite

### LINKED ISSUE #28355: blood-red magboots need an on icon
## Description
the opposite of #28354

### PR DESCRIPTION
**Pull request by dragonryan06**
## About the PR

Blood-red magboots would only update their game sprite and toggle action sprite when active, for the other actions they provided and for the item icon in the inventory they would continue to appear off. Fixed this!

Speed Boots would only update their item and action sprite, the clothing/held sprite would still appear off. Fixed this as well! Also left a boolean on the component they use for future boots of the sort (slowboots when).

## Why / Balance

- Fixes #28354
- Fixes #28355

## Technical details

The magboots fix was a simple yml mistake

For the speedboots I added a new bool to ToggleClothingSpeedComponent.cs regarding whether the entity's sprite would need updating after activation, and then implemented a check for this in ClothingSpeedModifierSystem.cs modeled after the way magboots update their sprites in MagbootsSystem.cs.

A future PR might want to look into combining some of these systems, there seems to be some redundancy but I didn't want to fix what isn't broken.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
 - fix: Blood-red Magboots and Speed Boots on/off lights now work as intended.

### CODE REVIEW

**Comment by Cojoke-dot** [comment]
#28653

File: Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml [hunk 1]
```diff
@@ -97,6 +97,13 @@
     walkModifier: 0.95
     sprintModifier: 0.9
     enabled: false
+  - type: Appearance
+  - type: GenericVisualizer
+    visuals:
+      enum.ToggleVisuals.Toggled:
+        enum.ToggleVisuals.Layer:
+          True: {state: icon-on}
+          False: {state: icon}
```
Comment (MilenVolf) [inline Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml:106]: This doesn't seem to fix anything. The core of the problem is in `Content.Client/JetpackSystem`. The case is that Jetpack and magboots use the same postfix `-on`, same with item icon.
Besides, `- type: Appearance` already inherited by the parent, so no need to duplicate it.

**Review by MilenVolf — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml (hunk 1 above)
Comment (dragonryan06) [reply Resources/Prototypes/Entities/Clothing/Shoes/magboots.yml:106]: My bad, I seem to have not gotten a notification over this reply. I am very new to the codebase but when I was poking around with this, I noticed that that was one difference between the boots and, upon duplicating it it seemed to work. You're saying that these lines don't do anything at all? If so, I could just remove that file from the PR...

**Comment by dragonryan06** [comment]
Since it seems i messed something up and #28653 got merged instead, I'm gonna close this. Sorry for the trouble lol!

**Pull request closed without merging**